src/settings/PatronNotices/PatronNoticeDetail.test.js

```diff
@@ -0,0 +1,312 @@
+    description: 'testDescription',
+    category: 'testCategory',
+  };
+  const defaultTestProps = {
+    initialValues: defaultInitialValues,
+  };
+  const keyValueCallOrderByPlace = {
+    noticeName: 1,
+    noticeActive: 2,
+    noticeDescription: 3,
+    noticeCategory: 4,
+    subject: 5,
+    body: 6,
+  };
+  const getById = (id) => within(screen.getByTestId(id));
+
+  afterEach(() => {
+    Accordion.mockClear();
+    AccordionSet.mockClear();
+    Button.mockClear();
+    KeyValue.mockClear();
+    PreviewModal.mockClear();
+    tokensReducer.mockClear();
+  });
+
+  describe('with default props', () => {
+    beforeEach(() => {
+      render(<PatronNoticeDetail {...defaultTestProps} />);
+    });
+
+    it('should render "notice name" KeyValue', () => {
+      expect(KeyValue).toHaveBeenNthCalledWith(
+        keyValueCallOrderByPlace.noticeName,
+        {
+          label: labelIds.noticeName,
+          value: defaultInitialValues.name,
+        }, {}
+      );
+    });
+
+    it('should render "notice description" KeyValue', () => {
+      expect(KeyValue).toHaveBeenNthCalledWith(
+        keyValueCallOrderByPlace.noticeDescription,
+        {
+          label: labelIds.noticeDescription,
+          value: defaultInitialValues.description,
+        }, {}
+      );
+    });
+
+    it('should render "notice category" KeyValue', () => {
+      expect(KeyValue).toHaveBeenNthCalledWith(
+        keyValueCallOrderByPlace.noticeCategory,
+        {
+          label: labelIds.noticeCategory,
+          value: defaultInitialValues.category,
+        }, {}
+      );
+    });
+
+    describe('AccordionSet', () => {
+      it('should render AccordionSet', () => {
+        expect(AccordionSet).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            accordionStatus: {
+              'email-template-detail': true,
+            },
+          }), {}
+        );
+      });
+
+      it('should toggle accordions', () => {
+        expect(AccordionSet).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            accordionStatus: {
+              'email-template-detail': true,
+            },
+          }), {}
+        );
+
+        const accordionSet = screen.getByTestId('accordionSet');
+        fireEvent.click(accordionSet);
+
+        expect(AccordionSet).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            accordionStatus: {
+              'email-template-detail': false,
+            },
+          }), {}
+        );
+      });
+    });
+
+    it('should render "email" Accordion', () => {
+      expect(Accordion).toHaveBeenLastCalledWith(
+        expect.objectContaining({
+          id: 'email-template-detail',
+          label: labelIds.email,
+        }), {}
+      );
+    });
+  });
+
+  [
+    false,
+    true,
+  ].forEach((active) => {
+    describe(`when 'active' is ${active}`, () => {
+      beforeEach(() => {
+        const props = {
+          ...defaultTestProps,
```

Comment:
In `defaultTestProps` we have only `initialValues`, but we redefine it below, so correct me if i wrong, but we don't need `defaultTestProps` here and in few places below.